### DISCUSSION TITLE
Extract SAP PDF Report which consit of vertical and horizontal line

### DISCUSSION THREAD

**Discussion by zbjit** [jsvine/pdfplumber#656] [Ask for help with specific PDFs]
Hello all,

I have a SAP PDF Report which consis of vertical and horizontal line, not real table,  I define the explicit_vertical_lines for horizontal_strategy to extract the PDF report to table, but all the horizontal line was included in the extract result.

The table_setting is :
`table_settings = {
    "vertical_strategy": "explicit",
     "horizontal_strategy": "text",
    "explicit_vertical_lines": [32, 76, 132, 384, 428, 448, 488, 532, 552, 596, 644, 684, 728, 768, 812, 835],
}
`

The result of debug_tablefinder of page 1:
![output](https://user-images.githubusercontent.com/74346670/169761923-7e36accf-6071-4dc8-815e-83b5f9af2568.png)

The output of extract_table:
<img width="316" alt="output2" src="https://user-images.githubusercontent.com/74346670/169762723-141ae1a8-6969-47db-a8d6-bdd77db3edf4.png">

As you saw, the horizontal line had been recognized as row and also output, could you please have any ideas to  
skip these lines? thank you.

Attached the PDF and notebook in the end.
[SAP_Report.pdf](https://github.com/jsvine/pdfplumber/files/8752345/SAP_Report.pdf)
[PDF_Table_Debug_5.ipynb.txt](https://github.com/jsvine/pdfplumber/files/8752354/PDF_Table_Debug_5.ipynb.txt)

**Comment by jsvine** [comment]
This is a very interesting PDF, @zbjit! Rather than use graphical lines, it seems to use [unicode box-drawing characters](https://en.wikipedia.org/wiki/Box-drawing_character). My suggestion would be to filter those out, with something like:

```python
box_chars = list("─┬┌┴") # This list is incomplete, you'll have to add the others
filtered = my_page.filter(lambda obj: not (obj.get("text") in box_chars))
table = filtered.extract_text(...)
```

**Reply by zbjit** [reply]
@jsvine Appreciate your help. As follow your guide, I got filtered page and extracted table, then fill in dataframe and dropna, in the final got expected table.
```
df = pd.DataFrame(table[0:], columns=columns)
df = df.replace('│', '', regex=True)
df = df.replace(r'^\s*$', np.nan, regex=True).dropna(axis=0, how='all')
```
Thank you again.
